anomalous-token-transfers-ml-py/README.md

```diff
@@ -0,0 +1,182 @@
+# Anomalous Transaction with Token Transfers
+
+## Description
+
+This bot utilizes machine learning to detect anomalous transactions with erc20 token transfers.
+
+### Machine Learning Model Description
+
+```
+IsolationForest(random_state=42, n_estimators=100)
+```
+
+An [Isolation Forest](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.IsolationForest.html#sklearn.ensemble.IsolationForest) model was trained to detect anomalous tx with erc20 token transfers. The model returns -1 for outliers and 1 for inliers, and the int values are mapped to string labels `ANOMALY` and `NORMAL` for human readability. The model considered 0.387% of the training dataset to be anomalous, including 4 of the following known exploit transactions:
+
+* (DETECTED) '0x2b023d65485c4bb68d781960c2196588d03b871dc9eb1c054f596b7ca6f7da56', # SaddleFinance Exploit
+* (DETECTED) '0xcd314668aaa9bbfebaf1a0bd2b6553d01dd58899c508d4729fa7311dc5d33ad7', # Beanstalk Flashloan Exploit
+* (DETECTED) '0xab486012f21be741c9e674ffda227e30518e8a1e37a5f1d58d0b0d41f6e76530', # FeiProtocol-Fuse Exploit
+* (DETECTED) '0x958236266991bc3fe3b77feaacea120f172c0708ad01c7a715b255f218f9313c', # Inverse Finance Exploit 2
+* '0x600373f67521324c8068cfd025f121a0843d57ec813411661b07edc5ff781842', # Inverse Finance Exploit 1
+* '0xe0b0c2672b760bef4e2851e91c69c8c0ad135c6987bbf1f43f5846d89e691428', # Revest Finance Exploit
+
+The model was trained on 14,469 transactions with the following 45 features:
```

Comment:
What is your recommendation around updating this model? Would it go stale and degrade over time?

Reply:
good question, I'll include that. I'm also thinking about including a description on tweaking the anomaly alerting threshold based on past predictions in production. maybe that can be a separate guide.

Reply:
Added description.